scripts/docker/dockerfile_gcc-8

```diff
@@ -1,25 +1,38 @@
 FROM axom/compilers:gcc-8
+LABEL maintainer="white238@llnl.gov"
+ARG branch=develop
 
+SHELL ["/bin/bash", "-c"]
 RUN sudo apt-get update -y
-RUN sudo apt-get install curl -fy
-
+RUN sudo apt-get install curl wget ssh pkg-config build-essential libopenblas-dev tar python3 -fy
 RUN sudo apt-get -qq install -y --no-install-recommends gfortran-7 && sudo update-alternatives --install /usr/bin/gfortran gfortran /usr/bin/gfortran-7 100
+RUN sudo ln -f -s python3 /usr/bin/python # Compatibility with legacy Python
 
-RUN git clone --recursive https://github.com/LLNL/axom.git
+# Avoid building CMake from scratch
+RUN wget https://cmake.org/files/v3.10/cmake-3.10.1-Linux-x86_64.sh
+RUN sudo sh cmake-3.10.1-Linux-x86_64.sh --prefix=/usr --exclude-subdir
 
-RUN git clone --recursive https://github.com/LLNL/RAJA.git
+# Avoid building all MPICH3.3 dependencies from scratch - is there a xenial-compatible binary somewhere?
+RUN wget http://www.mpich.org/static/downloads/3.3.2/mpich-3.3.2.tar.gz && tar xfz mpich-3.3.2.tar.gz
+RUN cd mpich-3.3.2 && ./configure -prefix=/usr && make -j && sudo make install
 
-WORKDIR "/home/axom/RAJA"
-RUN mkdir build
+# RUN sudo useradd -m -s /bin/bash -G sudo axom
 
-WORKDIR "/home/axom/RAJA/build"
-RUN cmake ../. -DENABLE_TESTS=OFF
-RUN make 
-RUN sudo make install
+WORKDIR "/home/axom"
+USER axom
 
-WORKDIR "/home/axom/axom"
-RUN ./scripts/uberenv/uberenv.py --spack-config-dir=./scripts/uberenv/spack_configs/docker --spec=%gcc@8.1.0 --prefix=/home/axom/axom_tpls -k
+RUN git clone --recursive --branch $branch --single-branch --depth 1 https://github.com/LLNL/axom.git
 
-WORKDIR "/home/axom"
+# New containers will use Python3
+RUN python3 ./axom/scripts/uberenv/uberenv.py --spack-config-dir=./axom/scripts/uberenv/spack_configs/docker/ubuntu16/ --project-json=./axom/scripts/uberenv/project.json \
+                                               --spec=%gcc@8.1.0 --prefix=/home/axom/serac_tpls -k
+
+# Make sure the new hostconfig worked
+RUN cd axom && python3 config-build.py -hc *.cmake -bp build && cd build && make -j8 VERBOSE=1 && make CTEST_OUTPUT_ON_FAILURE=1 test ARGS="-T Test -VV -j8"
```

Comment:
Minor: What is the name of the `hc` file?  
I think uberenv prints it, right?

Reply:
The host config is some oddly named file. For example, when I get the clang@10 host config from the workflow job  (it's an artifact), it is called `buildkitsandbox-linux-ubuntu18.04-x86_64-clang@10.0.0.cmake`.  I am not sure how stable that name is, specifically the "buildkitsandbox" part.

Reply:
It should be pretty stable - the Docker build action that's used here is versioned and will always use buildx/buildkit.  The wildcard was added for the "manual" docker builds, in which the hostconfig would be prefixed with the (unstable) hash of the Docker image.